pynest/nest/tests/test_parrot_neuron.py

```diff
@@ -0,0 +1,162 @@
+
+    def test_ParrotNeuronIgnoreSpike(self):
+        """Check parrot_neuron ignores spikes on port 1"""
+
+        # connect with arbitrary delay to port 1
+        nest.Connect(self.source, self.parrot, syn_spec={"receptor_type": 1})
+        nest.Simulate(3.)
+
+        # get spike from parrot neuron, assert it was ignored
+        events = nest.GetStatus(self.spikes)[0]["events"]
+        post_time = events['times'][events['senders'] == self.gid_parrot]
+        assert len(post_time) == 0, "Parrot neuron failed to ignore spike arriving on port 1"
+
+
+@nest.check_stack
+class ParrotNeuronSTDPTestCase(unittest.TestCase):
+    """
+    Check STDP protocol between two parrot_neurons connected by a stdp_synapse.
+    Exact pre- and post-synaptic spike times are set by spike_generators
+    connected to each parrot neuron. Additional spikes sent through the
+    stdp_synapse are explicitly ignored in the postsynaptic parrot_neuron
+    by setting the stdp_synapse to connect to port 1.
+    """
+
+    def test_ParrotNeuronSTDPProtocol(self):
+        """Check STDP between parrot_neurons changes weights, ignores transmitted spikes"""
+
+        nest.set_verbosity("M_WARNING")
+        nest.ResetKernel()
+
+        # set pre and postsynaptic spike times
+        dt = 5.
+        delay = 1.  # delay for connections
+
+        pre_times = [100., 300., 500., 700., 900.]
+        pre_times = [k-delay for k in pre_times]  # set the correct real spike times
+        post_times = [k+dt for k in pre_times]
+
+        # create spike_generators with these times
+        pre_spikes = nest.Create("spike_generator", 1, {"spike_times": pre_times})
+        post_spikes = nest.Create("spike_generator", 1, {"spike_times": post_times})
+
+        # create parrot neurons and connect spike_generators
+        pre_parrot = nest.Create("parrot_neuron", 1)
+        post_parrot = nest.Create("parrot_neuron", 1)
+        nest.Connect(pre_spikes, pre_parrot, syn_spec={"delay": delay})
+        nest.Connect(post_spikes, post_parrot, syn_spec={"delay": delay})
+
+        # create spike detector
+        spikes = nest.Create("spike_detector")
+        nest.Connect(post_parrot, spikes)
+
+        # connect both parrot neurons with a stdp synapse onto port 1
+        # thereby spikes transmitted through the stdp connection are
+        # not repeated postsynaptically.
+        syn_spec = {
+           "model": "stdp_synapse",
+           "receptor_type": 1,  # set receptor 1 postsynaptically, to not generate extra spikes
+        }
+        conn_spec = {
+           "rule": "one_to_one",
+        }
+        nest.Connect(pre_parrot, post_parrot, syn_spec=syn_spec, conn_spec=conn_spec)
+
+        # get STDP synapse and initial weight
+        syn = nest.GetConnections(source=pre_parrot, synapse_model="stdp_synapse")
+        initial_weight = nest.GetStatus(syn)[0]['weight']
+
+        nest.Simulate(1000)
+
+        # check that the STDP protocol has elicited a weight change.
```

Comment:
@flinz makes a good point about the proper place for testing the stdp_synapse being in the test for that synapse (which already exists). Adding this test would indeed be redundant. I think it would be sufficiently reassuring to check that the change of weight is in the expected direction.